### ISSUE TITLE
[Feature] Add extended official settlement nuance coverage

### ISSUE THREAD

**Issue by hnnng-w** [hnnng-w/skatmind#21]
The current settlement layer supports core single-game settlement, Suit/Grand overbid settlement, claim/concession point assignment, and partial ISkO-style single-game rating.

This issue tracks future improvements for more detailed official settlement nuances that are not yet fully modeled.

Scope:

Review official Skat settlement edge cases that are not currently covered.
Identify which cases should affect final_settlement_summary.
Add focused tests for each supported settlement nuance.
Keep final_settlement_summary separate from performance_rating_summary.
Avoid implementing full list/series/tournament rating in this issue.

Out of scope:

Full ISkO list aggregation.
Series or tournament scoring.
Four-player table support.
Machine-learning based decision quality.

Acceptance criteria:

Missing official settlement nuances are listed and prioritized.
At least one additional settlement nuance is implemented with tests.
Existing overbid, claim/concession, and normal settlement behavior remains backward-compatible.
Documentation is updated if output behavior changes.

**Comment by hnnng-w** [comment]
Implemented the first small settlement nuance slice for issue #21.

What changed:
- Added achieved Schneider settlement adjustment for completed non-null suit/grand games.
- `game_value_summary.game_value` remains the declared/pre-result value.
- `final_settlement_summary.effective_game_value` now adds one base-value level for final Schneider.
- Derived `settlement_score` updates from the adjusted effective value.
- Updated focused final-settlement tests and the `grand_complete_declarer_win` example expectations.
- Updated scoring/output docs and corrected overbid docs to reflect current `winner` plus `is_loss` behavior.

Changed areas:
- `src/skat_ai/final_settlement.py`
- `tests/test_final_settlement.py`
- `tests/test_examples.py`
- `docs/scoring.md`
- `docs/output_json.md`
- `docs/overbid.md`

Checks run:
- `.venv\Scripts\python.exe -m pytest tests/test_final_settlement.py`
- `.venv\Scripts\python.exe -m pytest tests/test_examples.py`
- `.venv\Scripts\python.exe -m pytest tests/test_overbid.py tests/test_game_result.py tests/test_performance_rating.py`
- `. .\.venv\Scripts\Activate.ps1; .\scripts\check.ps1`

Full check passed: yes, with the virtual environment active.
Product code changed: yes.

Recommendation: keep issue #21 open. This implements one additional official settlement nuance, but Schwarz, announcement-specific handling, and other settlement edge cases remain future slices.

**Comment by hnnng-w** [comment]
Implemented the next small issue #21 slice: Schneider announcement failure in final settlement.

Changed areas:
- `src/skat_ai/final_settlement.py`
- `tests/test_final_settlement.py`
- `docs/scoring.md`

Behavior added:
- Completed non-null suit/grand games now treat failed announced Schneider as a declarer settlement loss.
- The card-point `winner` remains unchanged.
- `effective_game_value` uses the declared game value that already includes the Schneider announcement level.
- Failed announcements do not add an additional achieved-Schneider level.
- Successful announced Schneider keeps the intended level counting by adding only the separate achieved-Schneider level.

Checks run:
- `.venv\Scripts\python.exe -m pytest tests/test_final_settlement.py`
- `.venv\Scripts\python.exe -m pytest tests/test_game_value.py tests/test_game_result.py tests/test_overbid.py`
- `.\scripts\check.ps1` with `.venv\Scripts` prepended to `PATH`

Full check passed:
- Ruff passed
- input schema validation passed
- generated output schema validation passed
- pytest passed with 797 tests

Product code changed: yes.

No schema, CLI, game result, overbid, game value, or example changes were made.

Recommendation: keep issue #21 open for remaining official settlement nuances.

**Comment by hnnng-w** [comment]
Implemented a final regression slice for issue #21 covering overbid and achieved Schneider precedence.

Changed files:

* `tests/test_final_settlement.py`

What changed:

* Added coverage proving that for supported Suit/Grand overbid losses, `overbid_summary.required_game_value` remains the authoritative `final_settlement_summary.effective_game_value`.
* Confirmed achieved Schneider does not add another base-value level on top of the required overbid value.
* Confirmed overbid still forces `is_loss = true` while the card-point `winner` remains unchanged.

Checks:

* `.venv\Scripts\python.exe -m pytest tests/test_final_settlement.py` passed: 26 passed.
* `.venv\Scripts\python.exe -m pytest tests/test_overbid.py` passed: 12 passed.
* `.\scripts\check.ps1` passed: Ruff, schema validation, generated output validation, and 798 tests passed.

Product code changed: no. Test-only slice.

Issue summary:

* Achieved Schneider settlement is implemented.
* Failed announced Schneider settlement is implemented.
* Overbid and achieved Schneider precedence is covered.
* Related tests and documentation have been updated.

Recommendation: close issue #21. Remaining Schwarz and Schwarz-announcement behavior should be handled in a separate focused issue once trick-based Schwarz detection can be modeled safely.

**Comment by hnnng-w** [comment]
Read-only backlog audit finding:

Issue #21 has met its stated acceptance slice. The current implementation covers achieved Schneider settlement adjustment, failed Schneider announcement settlement loss, and overbid plus achieved Schneider precedence. Focused tests and documentation exist for the implemented behavior.

Remaining official settlement work, especially Schwarz and Schwarz-announcement handling, should be tracked in a separate focused issue rather than keeping this broad umbrella open.

Changed files in the completed implementation are already in the repository history. No product code was changed by this audit.

Checks run for this audit: none, read-only audit only.

Recommendation: close #21 as completed.
